### DISCUSSION TITLE
Importing Landmark (mrk.json) files

### DISCUSSION THREAD

**Discussion by amygutierrez** [Project-MONAI/MONAILabel#1659] [Q&A]
I am trying to import a landmarks file that contains coordinates of 3 points. This file is a mrk.json file that is exported from 3DSlicer. I am trying to make these landmarks files import similarly to how segmentation files are also imported. My directory looks like this:
where `final` contains segmentation files and `landmarks` has the landmarks mrk.json files. 
```
D:\DATA
└───labels
    ├───final
    ├───landmarks
    └───original
```
The segmentation import works well, and I have used the same process with the landmarks file:
 https://github.com/amygutierrez/MONAILabel/blob/df6032d7b30b8b85ff13abb443b47c33047bfb5a/plugins/slicer/MONAILabelReviewer/MONAILabelReviewer.py#L1718-L1726

I have also changes the corresponding code in ImageDataController and MONAIServerREST. 

[ImageDataController.py](https://github.com/amygutierrez/MONAILabel/blob/df6032d7b30b8b85ff13abb443b47c33047bfb5a/plugins/slicer/MONAILabelReviewer/MONAILabelReviewerLib/ImageDataController.py#L237-L247)
```
    def requestPointList(self, image_id: str, tag: str) -> requests.models.Response:
        print(image_id)
        print(tag)
        img_blob = self.monaiServerREST.requestPointList(image_id, tag)
        print(img_blob)
        logging.info(
            "{}: Segmentation successfully requested from MONAIServer (image id: {})".format(
                self.getCurrentTime(), image_id
            )
        )
        return img_blob
```

[MonaiServerREST.py](https://github.com/amygutierrez/MONAILabel/blob/df6032d7b30b8b85ff13abb443b47c33047bfb5a/plugins/slicer/MONAILabelReviewer/MONAILabelReviewerLib/MonaiServerREST.py#L60-L87)
```
    def requestPointList(self, image_id: str, tag: str) -> requests.models.Response:
        # if tag == "":
        print(tag)
        #tag = "landmarks"
        download_uri = f"{self.serverUrl}/datastore/label?label={quote_plus(image_id)}&tag={quote_plus(tag)}"
        print(f'point list download_uri: {download_uri}')
        logging.info(f"{self.getCurrentTime()}: REST: request segmentation '{download_uri}'")


        try:
            print(f'trying pointlist request')
            response = requests.get(download_uri, timeout=5)
            print(f'pointlist response: {response}')
        except Exception as exception:
            logging.warning(
                "{}: Segmentation request (image id: '{}') failed due to '{}'".format(
                    self.getCurrentTime(), image_id, exception
                )
            )
            return None
        if response.status_code != 200:
            logging.warn(
                "{}: Segmentation request (image id: '{}') failed due to response code: '{}'".format(
                    self.getCurrentTime(), image_id, response.status_code
                )
            )
            return None


        return response
```

I also added the appropriate file extension in data/local.py:
[datasource/local.py](https://github.com/amygutierrez/MONAILabel/blob/23e22e81d063ce0de9b3bf1cb3a49bc1328bdfda/monailabel/datastore/local.py#L105)
```
        extensions=("*.nii.gz", "*.nii", "*.mrk.json", "*.mrk", "*.json"),
```
The download_uri for the landmarks file ends up looking like this: 
The file name is123456_OSTIUM.mrk.json
```
http://localhost:8080/datastore/label?label=123456_OSTIUM&tag=landmarks
```
For reference, the segmentation download_uri looks like this: 
The file name is 123456_0000.nii.gz
```
http://localhost:8080/datastore/label?label=123456_0000&tag=final
```

The issue I'm experiencing is that the response I get from the request, I get a response of 404. In other words, this comes out as `None`
`requests.get(download_uri, timeout=5)`

My suspicion is that the download_uri is not pointing at the correct location, or it is not configured correctly. Would appreciate any advice on what I am doing wrong. 
Thank you!!

**Comment by diazandr3s** [comment]
Hi @amygutierrez,

Thanks for opening this discussion.
How are you planning to use MONAI Label Reviewer?  Did you mean the [MONAI Label plugin](https://github.com/amygutierrez/MONAILabel/blob/df6032d7b30b8b85ff13abb443b47c33047bfb5a/plugins/slicer/MONAILabel/MONAILabel.py)?

Please let us know,

**Comment by amygutierrez** [comment]
Hello, 
Thanks for taking the time and looking into this! Yes, I am using the slicer plugin code. But all the changes I've made are mainly the MONAILabelReviewer.py script, for our specific use case. 
Basically when I import the module into 3D Slicer, the path I am using is /MONAILabel/plugins/slicer/MONAILabelReviewer and then I use the MONAILabel Reviewer module for segmentation annotation.

**Reply by diazandr3s** [reply]
Hi,

Is there a reason to use MONAILabelReviewer instead of MONAILabel plugin only?
